### PR TITLE
Adjust QueueManager to use dcicutils.misc_utils.RateManager

### PR DESCRIPTION
**Pull request by netsettler**
This implements the purge lockout by using the new RateManager class.

### CODE REVIEW

File: snovault/elasticsearch/indexer_queue.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def dlq_to_primary(context, request):
     return response
 
 
-class QueueManager(object):
+class QueueManager (object):
```
Comment (willronchetti) [inline snovault/elasticsearch/indexer_queue.py:181]: This space has no effect? Seems weird to have.

File: snovault/elasticsearch/indexer_queue.py [hunk 2]
```diff
@@ -435,24 +422,7 @@ def initialize(self, dlq=False):
...
-                              % (self.purge_queue_timestamp, seconds_since_last_purge))
-            if self.PURGE_QUEUE_SLEEP_FOR_SAFETY:
-                action_message = "Waiting %s seconds before attempting another." % wait_seconds
-                log.warning("%s %s" % (shared_message, action_message))
-                time.sleep(wait_seconds)
-            else:
-                action_message = "Continuing anyway because QueueManager.PURGE_QUEUE_SLEEP_FOR_SAFETY is False."
-                log.warning("%s %s" % (shared_message, action_message))
-        self.purge_queue_timestamp = datetime.datetime.now()
+        self.rate_manager.wait_if_needed()
```
Comment (willronchetti) [inline snovault/elasticsearch/indexer_queue.py:425]: Much nicer to do!

File: snovault/elasticsearch/indexer_queue.py (hunk 1 above)
Comment (netsettler) [reply snovault/elasticsearch/indexer_queue.py:181]: I probably leaned on the keyboard at the wrong time. :)

**Comment by netsettler** [comment]
> Make sure before you merge this that you test it out with FF/CGAP to be sure it doesn't block unnecessarily, especially in the local deployment

I did Travis runs on CGAP and FF but did not deploy them. I can do a local deploy to try it out, but I'll want that deploy1 bug ([C4-233](https://hms-dbmi.atlassian.net/browse/C4-233)) fixed. :) I'll talk to @ebberg about that.

**Comment by netsettler** [comment]
> I can do a local deploy to try it out, but I'll want that deploy1 bug ([C4-233](https://hms-dbmi.atlassian.net/browse/C4-233)) fixed. :) I'll talk to @ebberg about that.

OK, so that bug is fixed (thanks, @ebberg) and I ran deploy1/deploy2 on both FF and CGAP and it went smoothly.

I'll upload a fix to the errant Space character that got inserted, and then merge this. :)

**Pull request merged**